aiida/engine/processes/exit_code.py

```diff
@@ -8,49 +8,66 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """A namedtuple and namespace for ExitCodes that can be used to exit from Processes."""
-
-from collections import namedtuple
-
 from aiida.common.extendeddicts import AttributeDict
 
 __all__ = ('ExitCode', 'ExitCodesNamespace')
 
-ExitCode = namedtuple('ExitCode', ['status', 'message', 'invalidates_cache'])
-ExitCode.__new__.__defaults__ = (0, None, False)
-"""
-A namedtuple to define an exit code for a :class:`~aiida.engine.processes.process.Process`.
 
-When this namedtuple is returned from a Process._run() call, it will be interpreted that the Process
-should be terminated and that the exit status and message of the namedtuple should be set to the
-corresponding attributes of the node.
+class ExitCode:
```

Comment:
I'm not entirely convinced we should mix the "exit code" and "exit code template" concepts in the same class. On the one hand, this makes it easy in the `spec.exit_code` call (you just pass on the message) -- but on the other hand, this makes it almost impossible for the code that uses the exit code to figure out which one it is.

Then again I can't come up with a convincing example for when you would need to distinguish this, so this might be fine.

Reply:
The template part here only really applies to the message though and as the message really is just a human readable description/alternative of the exit status, I don't think this is a problem. Any machine will only ever just have to look at the status which is "immutable" and should never even look at the message.

Reply:
Yeah, by "calling" code I meant where the exit code is returned from.. like, maybe you want to know if you should "format" the exit code or no.

Reply:
But, again, I'm not super convinced on this. Since the message has to use named formatting you can always "format" with a dict that has too many keys.